frame/frame.go

```diff
@@ -0,0 +1,181 @@
+package frame
+
+import (
+	"bytes"
+	"encoding/hex"
+	"fmt"
+	"github.com/datastax/go-cassandra-native-protocol/message"
+	"github.com/datastax/go-cassandra-native-protocol/primitive"
+)
+
+// A high-level representation of a frame, where the body is fully decoded.
+type Frame struct {
+	Header *Header
+	Body   *Body
+}
+
+// A low-level representation of a frame, where the body is not decoded.
+type RawFrame struct {
+	Header *Header
+	Body   RawBody
```

Comment:
Didn't we discuss this in a previous PR and decided to change the name of the fields instead of adding a type alias? I don't like this type alias, it just hides the actual type imo.

Reply:
I actually suggested a `type RawBody struct { Body: []byte }` and we abandoned the idea, I didn't know we also ruled out the type alias approach. But no problem I will remove it, I don't have any preference here.

Reply:
oh I misunderstood then, but in any case I think it makes it more clear to have a different field name even if we keep the type alias.

Reply:
Yep, the more it goes and the more I agree that `[]byte` is indeed perfect here, sorry for being so reluctant initially.